x/model/keeper/msg_server_model.go

```diff
@@ -149,7 +149,7 @@ func (k msgServer) UpdateModel(goCtx context.Context, msg *types.MsgUpdateModel)
 	}
 
 	model.EnhancedSetupFlowOptions = msg.EnhancedSetupFlowOptions
-	if msg.EnhancedSetupFlowOptions == 0 {
+	if msg.EnhancedSetupFlowOptions&1 == 1 {
```

Comment:
I think this logic is not correct. msg.EnhancedSetupFlowOptions may not be present in the Update command, meaning that the old value needs to be preserved. On the other hands, only some of the EndhancedSetupFlow values may be present (needs to be updated)

I think the right logic would be the following: 
- get the new values for all EnhancedSetupFlow fields (old one if not in the msg or updated one if in the msg)
- if the new value of EnhancedSetupFlowOptions&1 == 1, make sure all new values for other corresponding  EnhancedSetupFlow are set
- if the new value of EnhancedSetupFlowOptions&1 != 1, make sure all new values for other corresponding  EnhancedSetupFlow are NOT set

We may think how to simplify that logic...

Reply:
Even if we decide to do https://github.com/zigbee-alliance/distributed-compliance-ledger/pull/588#discussion_r1702084104, I think the logic under `else` here is not correct. If Enhanced* fields are not provided, then we should keep old/current values, not override them to empty ones.